### ISSUE TITLE
Vector inputs are not resortored upon compute

### ISSUE THREAD

**Issue by dgrechka** [itislab/alpheus#78]
I get 
```
(venv) PS D:\bengaliai-cv19> alpheus compute .\experiment_outputs\dgrechka_2_mobileNetV2_bottleneck\vector.alph
System.ArgumentException: Index doesn't correspond to the rank of the artefact
Parameter name: index
   at ItisLab.Alpheus.DependencyGraph.LinkToArtefact.AnalyzeStatus(FSharpFunc`2 checkStoragePresence, FSharpList`1 index) in C:\repos\alpheus\AlpheusCore\DependencyGraph.fs:line 248
   at ItisLab.Alpheus.API.statuses@377.Invoke(FSharpList`1 idx) in C:\repos\alpheus\AlpheusCore\API.fs:line 377
   at Microsoft.FSharp.Collections.Internal.IEnumerator.map@75.DoMoveNext(b& curr) in E:\A\_work\130\s\src\fsharp\FSharp.Core\seq.fs:line 78
   at Microsoft.FSharp.Collections.Internal.IEnumerator.MapEnumerator`1.System-Collections-IEnumerator-MoveNext() in E:\A\_work\130\s\src\fsharp\FSharp.Core\seq.fs:line 64
   at System.Collections.Generic.List`1.AddEnumerable(IEnumerable`1 enumerable)
   at Microsoft.FSharp.Collections.SeqModule.ToArray[T](IEnumerable`1 source) in E:\A\_work\130\s\src\fsharp\FSharp.Core\seq.fs:line 823
   at <StartupCode$FSharp-Core>.$Async.Parallel@1175.Invoke(AsyncActivation`1 ctxt) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 1178
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 349
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronouslyInCurrentThread[a](CancellationToken cancellationToken, FSharpAsync`1 computation) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 870
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 890
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 1151
   at ItisLab.Alpheus.API.presenceCheckResult@370-2.Invoke(ArtefactVertex _arg4) in C:\repos\alpheus\AlpheusCore\API.fs:line 375
   at ItisLab.Alpheus.Results.ResultBuilder.Bind[a2,a3,a4](FSharpResult`2 m, FSharpFunc`2 f) in C:\repos\alpheus\AlpheusCore\ResultBuilder.fs:line 16
   at ItisLab.Alpheus.API.presenceCheckResult@369-1.Invoke(Unit _arg3) in C:\repos\alpheus\AlpheusCore\API.fs:line 369
   at ItisLab.Alpheus.Results.ResultBuilder.Bind[a2,a3,a4](FSharpResult`2 m, FSharpFunc`2 f) in C:\repos\alpheus\AlpheusCore\ResultBuilder.fs:line 16
   at ItisLab.Alpheus.API.presenceCheckResult@365.Invoke(Unit unitVar) in C:\repos\alpheus\AlpheusCore\API.fs:line 365
   at ItisLab.Alpheus.Results.ResultBuilder.Run[m](FSharpFunc`2 f) in C:\repos\alpheus\AlpheusCore\ResultBuilder.fs:line 31
   at ItisLab.Alpheus.API.compute@341-1.Invoke(Graph _arg1) in C:\repos\alpheus\AlpheusCore\API.fs:line 389
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvokeNoHijackCheck[a,b](AsyncActivation`1 ctxt, FSharpFunc`2 userCode, b result1) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 417
   at ItisLab.Alpheus.API.buildDependencyGraphAsync@22-3.Invoke(AsyncActivation`1 ctxt) in C:\repos\alpheus\AlpheusCore\API.fs:line 22
   at ItisLab.Alpheus.API.compute@339-3.Invoke(AsyncActivation`1 ctxt) in C:\repos\alpheus\AlpheusCore\API.fs:line 339
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 109
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 349
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronouslyInCurrentThread[a](CancellationToken cancellationToken, FSharpAsync`1 computation) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 870
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 890
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 1153
   at ItisLab.Alpheus.API.compute(String experimentRoot, ArtefactId artefactId) in C:\repos\alpheus\AlpheusCore\API.fs:line 338
   at ItisLab.Alpheus.CliRunner.run@66-25.Invoke(Tuple`2 _arg8) in C:\repos\alpheus\Alpheus.CLI\Runner.fs:line 66
   at ItisLab.Alpheus.Results.ResultBuilder.Bind[a2,a3,a4](FSharpResult`2 m, FSharpFunc`2 f) in C:\repos\alpheus\AlpheusCore\ResultBuilder.fs:line 16
   at ItisLab.Alpheus.CliRunner.run@63-24.Invoke(Unit unitVar) in C:\repos\alpheus\Alpheus.CLI\Runner.fs:line 65
   at ItisLab.Alpheus.Results.ResultBuilder.Run[m](FSharpFunc`2 f) in C:\repos\alpheus\AlpheusCore\ResultBuilder.fs:line 31
   at ItisLab.Alpheus.CliRunner.run(String programName, String workingDir, ParseResults`1 parseResults) in C:\repos\alpheus\Alpheus.CLI\Runner.fs:line 62
   at ItisLab.Alpheus.Program.main(String[] argv) in C:\repos\alpheus\Alpheus.CLI\Program.fs:line 42
```

while I have inputs (also vector) available in storage

Alph file:
```
{
  "FileFormatVersion": 1,
  "Origin": {
    "Case": "CommandOrigin",
    "Fields": [
      {
        "Inputs": [
          {
            "RelativePath": "..\\..\\code\\dgrechka\\train_mobileNetV2_bottleneck.py",
            "Hash": null
          },
          {
            "RelativePath": "..\\..\\data\\bengaliai-cv19\\",
            "Hash": null
          },
          {
            "RelativePath": "..\\..\\data\\5foldCvSplits\\*.val_ids.csv",
            "Hash": null
          }
        ],
        "Outputs": [
          {
            "RelativePath": "*\\",
            "Hash": null
          }
        ],
        "ResourceGroups": [
          "gpu"
        ],
        "SuccessfulExitCodes": [
          0
        ],
        "OutputIndex": 0,
        "WorkingDirectory": "..\\..",
        "Command": "python $in1 $in2 $in3 $out1",
        "Signature": "F6CAA8BF1432F2612D8F606B1A3636E90D52E58D",
        "OutputsCleanDisabled": false
      }
    ]
  },
  "IsTracked": false
}
```

**Comment by dgrechka** [comment]
after manual explicit restore of vectored input (e.g. `alpheus restore alpheus restore .\data\5foldCvSplits\vector.val_ids.csv.alph`) the compute started

**Comment by dgrechka** [comment]
Fixed in PR #87
